src/connectors/impls/elastic.rs

```diff
@@ -703,101 +704,156 @@ impl<'a, 'value> ESMeta<'a, 'value> {
+        }
+        if let Some(routing) = self.get_routing() {
+            op = op.routing(routing);
+        }
+        if let Some(pipeline) = self.get_pipeline() {
+            op = op.pipeline(pipeline);
+        }
+        ops.push(op).map_err(Error::from)?;
+        Ok(())
+    }
+
+    fn insert_delete_op(&self, ops: &mut BulkOperations) -> Result<()> {
+        let mut op: BulkDeleteOperation<()> = self
+            .get_id()
+            .map(BulkOperation::delete)
+            .ok_or_else(|| Error::from(Self::MISSING_ID))?;
+        if let Some(index) = self.get_index() {
+            op = op.index(index);
+        }
+        if let Some(version_type) = self.get_version_type() {
+            op = op.version_type(version_type);
+        }
+        if let Some(version) = self.get_version() {
+            op = op.version(version);
+        }
+        if let Some(if_primary_term) = self.get_if_primary_term() {
+            op = op.if_primary_term(if_primary_term);
+        }
+        if let Some(if_seq_no) = self.get_if_seq_no() {
+            op = op.if_seq_no(if_seq_no);
+        }
+        if let Some(routing) = self.get_routing() {
+            op = op.routing(routing);
+        }
+
+        ops.push(op).map_err(Error::from)?;
+        Ok(())
+    }
+
+    fn insert_create_op(&self, data: &Value, ops: &mut BulkOperations) -> Result<()> {
+        // create requires an `_id` here, which is not according to spec
+        // Actually `_id` should be completely optional here
+        // See: https://github.com/elastic/elasticsearch-rs/issues/190
+        let mut op = self
+            .get_id()
+            .map(|id| BulkOperation::create(id, data))
+            .ok_or_else(|| Error::from(Self::MISSING_ID))?;
+        if let Some(index) = self.get_index() {
+            op = op.index(index);
+        }
+        if let Some(pipeline) = self.get_pipeline() {
+            op = op.pipeline(pipeline);
+        }
+        if let Some(routing) = self.get_routing() {
+            op = op.routing(routing);
+        }
+        ops.push(op).map_err(Error::from)?;
+        Ok(())
+    }
+
+    // avoid a `.clone_static()` for the `raw` case
+    fn insert_raw_update_op(&self, data: &Value, ops: &mut BulkOperations) -> Result<()> {
+        let mut op = self
+            .get_id()
+            .map(|id| BulkOperation::update(id, data))
+            .ok_or_else(|| Error::from(Self::MISSING_ID))?;
+
+        op = self.apply_update_params(op);
+        ops.push(op).map_err(Error::from)?;
+        Ok(())
+    }
+
+    fn insert_update_op(&self, data: &Value, ops: &mut BulkOperations) -> Result<()> {
+        let mut op = self
+            .get_id()
+            .map(|id| {
+                // we are sure that this BulkUpdateOperation lives only as long as the event
+                // so it is fine to only do a `.clone()` here instead of a `.clone_static()`
+                let src = literal!({ "doc": data.clone() });
```

Comment:
this one is indeed serialized as soon as it is pushed below, so this is *totally* safe :P